Fix issue #28729: Functions default to commutative=True

jhaayushkumar opened the pull request:
Fixes #28729

#### Brief description of what is fixed or changed

Fixed bug where `0**(1 - F(x < 1))` evaluated to [nan](cci:1://file:///Users/ayushkumarjha/sympy-1/sympy/core/tests/test_relational.py:578:0-589:54) after substitution but stayed symbolic with direct construction. 
Functions now default to `commutative=True` like Symbols. This fixes the issue where [Function](cci:2://file:///Users/ayushkumarjha/sympy-1/sympy/core/function.py:381:0-814:23) objects with `Relational` arguments had `is_commutative=None`, which incorrectly propagated as `False` and caused [nan](cci:1://file:///Users/ayushkumarjha/sympy-1/sympy/core/tests/test_relational.py:578:0-589:54) in power operations.



#### Release Notes

<!-- BEGIN RELEASE NOTES -->
* core
  * Fixed bug where `0**(1 - F(x < 1))` incorrectly evaluated to [nan](cci:1://file:///Users/ayushkumarjha/sympy-1/sympy/core/tests/test_relational.py:578:0-589:54) after substitution. Functions now correctly default to `commutative=True`.
<!-- END RELEASE NOTES -->

oscarbenjamin commented on sympy/core/function.py:
```diff
@@ -614,7 +614,7 @@ def _eval_derivative(self, s):
 
     def _eval_is_commutative(self):
-        return fuzzy_and(a.is_commutative for a in self.args)
+        return True
```
This method should be removed. The idea should be that someone can do `Function('f', commutative=False)`.

jhaayushkumar committed:
Fix #28729: Make Functions default to commutative=True

Functions now inherit commutative=True by default, preventing expressions like 0**(1 - F(x < 1)) from incorrectly evaluating to nan when F has Relational arguments.
